docs/design/chart_format.md

```diff
@@ -0,0 +1,664 @@
+
+
+The following URI schemes must be supported:
+* http:
+* https:
+
+
+#### Special Form Google Storage and S3
+The [Google Storage scheme](https://cloud.google.com/storage/docs/interoperability) (`gs://`)  and the S3 scheme (`s3://`) can also be support. These forms do not follow the URL standard (bucket name is placed in the host field), but their behavior is well-documented.
+
+
+Examples:
+```
+gs://charts/nginx-1.2.3.tgz
+s3://charts/nginx-1.2.3.tgz
+```
+### Short Form
+A short form package reference provides the minimum necessary information for accessing a chart. Short form URLs use a defined [URI scheme](https://tools.ietf.org/html/rfc3986#section-3.1) of `helm`:.
+
+
+A generic short form maps to an object storage reference (`DOMAIN/REPOSITORY/RELEASE`).
+
+```
+helm:example.com/charts/nginx-1.2.3.tgz
+```
+
+For the purpose of providing versioning ranges, and also for backward compatibility, version requirements may be treated as as suffix condition, instead of part of the path:
+
+```
+helm:example.com/charts/nginx#1.2.3   // Exact version
+helm:example.com/charts/nginx#~2.1    // Version range
+helm:example.com/charts/nginx         // Latest release
+```
+
+The first of the above two short names is equivalent to helm:example.com/charts/nginx-1.2.3.tgz. The second example uses a semantic version filter ~1.2, which means “>=1.2.0, <1.3.0”, or “any patch version of 1.2”. Other filters include the “^” operator (^1 is “any minor version in the 1.x line), and “>”, “>=”, “=”, “<”, and “<=” operators. These are standard SemVer filter operators.
+
+
+Any short form handler should be able to resolve the default short form as specified above.
+
+
+### Local References
+During chart development, and in other special circumstances, it may be desirable to work with an unversioned, unpackaged local copy of a chart. For the sake of consistency across products, an explicit naming formula is to be followed.
+
+
+In cases where a local path is used in lieu of a full or short name, the path string _must_ begin with either a dot (.) or a slash (/) or the file schema(`file`:).
+
+
+Legal examples of this include:
+```
+  ./foo
+  ../foo
+  /foo
+  //foo
+  /.foo
+  file:///example/foo
+```
+
+Unprefixed relative paths are not valid. For example, `foo/` is not allowed as a local path, as it conflicts with a legal short name, and is thus ambiguous.
+
+
+
+## Appendix A: User Stories for Charts
+Personas:
+
+
+- Operator: Responsible for running an application in production.
+- Service Dev: Responsible for developing new definitions
+- App Dev: Developer who creates applications that make use of existing definitions, but does not create definitions.
+
+
+Stories:
+
+
+- As an operator, I want a build that is 100% reproducible (exact versions)
+- As an app dev, I want to be able to search for definitions...
+        * by keyword, where one app may have multiple keywords (e.g. Redis has storage, message queue)
+        * by name, where name may be "fuzzy"
```

Comment:
@michelleN yes, it does. Thanks for clarifying.

I'd suggest that we make clear in the text that we're referring to the chart name, using a document link to point back to its definition, so that the meaning is clear.